### ISSUE TITLE
bug(rust/gui-client/windows): UDP sockets fill up on Windows

### ISSUE THREAD

**Issue by jamilbk** [CloudFire-LLC/cloudfire#6454]
Seen in user's logs. Have not tried to reproduce yet.

```
[2m2024-08-27T12:47:16.676595Z[0m [33m WARN[0m [2mquinn_udp[0m[2m:[0m sendmsg error: Os { code: 10055, kind: Uncategorized, message: "An operation on a socket could not be performed because the system lacked sufficient buffer space or because a queue was full." }, Transmit: { destination: 50.170.132.214:54416, src_ip: Some(192.168.31.135), enc: None, len: 72, segment_size: None }
```

**Comment by jamilbk** [comment]
cc @conectado in case maybe the recent quinn bump caused it? #5872 #6314

**Comment by jamilbk** [comment]
Dropping to medium because not sure if / how this impacts operation.

**Comment by thomaseizinger** [comment]
Likely not a big issue, the OS will retry the query I assume.

**Comment by ReactorScram** [comment]
Isn't it strange that it would fill up at all though? If the system isn't under load

**Comment by thomaseizinger** [comment]
> Isn't it strange that it would fill up at all though? If the system isn't under load

Yeah it is strange. I have some long-term plans to get rid of our buffering in connlib's event loop and simply suspend until we can send a certain datagram. That should get rid of these things but it is not as easy I think.

**Comment by ReactorScram** [comment]
I mean, I haven't seen it in my logs lately. @jamilbk Was this the same user and same time they couldn't access DNS Resources?

**Comment by jamilbk** [comment]
Same user. I don't have info on the time it occurred though. Only that it was happening for most of his Windows fleet.

This customer is now in our Slack support so I'll ask for a more precise log if it happens again.

**Comment by thomaseizinger** [comment]
Pretty sure this is outdated, we now have separate threads for the UDP socket that suspend if we can't continue writing.
